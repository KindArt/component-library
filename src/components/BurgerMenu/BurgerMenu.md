Standard:

```jsx padded
let isOpen = false;
<BurgerMenu isOpen={isOpen} />;
```
