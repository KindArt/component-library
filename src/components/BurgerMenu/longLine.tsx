import React from 'react';

const LongLine = () => (
  <g id="Icon/20px/menu-[Extra]" transform="translate(0.000000, -9.000000)">
    <polygon id="Path" points="0 11 20 11 20 9 0 9" />
  </g>
);

export default LongLine;
