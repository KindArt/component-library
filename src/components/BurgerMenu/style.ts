import styled, { css } from 'styled-components';
import { Colours } from '../../constants/styles';

interface IWrapper {
  isOpen?: boolean;
}

export const BurgerMenuWrapper = styled.div<IWrapper>`
  height: 20px;
  width: 20px;
  position: relative;
  color: ${Colours.TEXT_SECONDARY};

  ${({ isOpen }) => isOpen && css``}
`;

interface ILongLineSVG {
  isOpen: boolean;
}

export const TopLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  left: 0;
  top: 2px;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;

      right: auto;
      left: 3px;
      top: 9px;

      transform: rotate(-45deg);
    `}
`;

export const MiddleLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  top: 9px;
  left: 0;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;

      opacity: 0;
    `}
`;

export const BottomLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  left: 0;
  top: 16px;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;
      right: auto;
      left: 3px;
      top: 9px;

      transform: rotate(45deg);
    `}
`;
