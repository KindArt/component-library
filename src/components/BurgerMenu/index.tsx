import React, { FC } from 'react';
import { BurgerMenuWrapper, TopLine, MiddleLine, BottomLine } from './style';
import LongLine from './longLine';

interface IBurgerMenu {
  className?: string;
  isOpen: boolean;
}

const BurgerMenu: FC<IBurgerMenu> = ({ className, isOpen }) => (
  <BurgerMenuWrapper className={className}>
    <TopLine fill="currentColor" height="2px" version="1.1" viewBox="0 0 20 2" width="20px" isOpen={isOpen}>
      <LongLine />
    </TopLine>
    <MiddleLine fill="currentColor" height="2px" version="1.1" viewBox="0 0 20 2" width="20px" isOpen={isOpen}>
      <LongLine />
    </MiddleLine>
    <BottomLine fill="currentColor" height="2px" version="1.1" viewBox="0 0 20 2" width="20px" isOpen={isOpen}>
      <LongLine />
    </BottomLine>
  </BurgerMenuWrapper>
);

export default BurgerMenu;
