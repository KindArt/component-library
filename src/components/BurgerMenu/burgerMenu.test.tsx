import React from 'react';
import { shallow } from 'enzyme';
import BurgerMenu from '.';

describe('<BurgerMenu />', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(<BurgerMenu isOpen={false} />);
  });
  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
