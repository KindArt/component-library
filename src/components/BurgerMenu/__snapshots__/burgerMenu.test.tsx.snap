// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerMenu /> should render 1`] = `
<styled.div>
  <styled.svg
    fill="currentColor"
    height="2px"
    isOpen={false}
    version="1.1"
    viewBox="0 0 20 2"
    width="20px"
  >
    <LongLine />
  </styled.svg>
  <styled.svg
    fill="currentColor"
    height="2px"
    isOpen={false}
    version="1.1"
    viewBox="0 0 20 2"
    width="20px"
  >
    <LongLine />
  </styled.svg>
  <styled.svg
    fill="currentColor"
    height="2px"
    isOpen={false}
    version="1.1"
    viewBox="0 0 20 2"
    width="20px"
  >
    <LongLine />
  </styled.svg>
</styled.div>
`;
